### ISSUE TITLE
ATR File memory leakage

### ISSUE THREAD

**Issue by djdduty** [TurokMods/T4Editor#8]
When an instance of opent4::ATRFile is loaded, it doesn't seem to be freeing all the memory that it allocated during load / initialization. If you keep loading ATR scenes and then deleting them the memory usage continues to rise and never frees.

It seems as though the cause may be that m_Data is not freed in the ATRFile destructor, Block's destructor does not free children blocks, and the m_Root Block is not freed in the ATRFile destructor either, nothing else seems to be freed there either.

Basically, selecting a level to load, and then selecting the exact same level to load repeatedly over and over should not be raising the allocated memory amount of the program (it might raise the first time, but then that memory will be freed and re-used by the application each time after).

**Comment by mdecicco** [comment]
I think I got the loader to delete all of the allocated data, but I can't launch the xcode profiling tools to verify that... @djdduty maybe you'll have more luck with that since you have a real mac. You may need to place your config.txt in ~/ for that to work, but that didn't fix it for me.
